### DISCUSSION TITLE
C# 12 `ref readonly` parameters support

### DISCUSSION THREAD

**Discussion by aalmada** [dotnet/docfx#9339] [General]
C# 12 introduces [`ref readonly` parameters](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-12.0/ref-readonly-parameters.md) support. 

Currently, its use [results in an error](https://github.com/NetFabric/NetFabric.Numerics/actions/runs/6626235079/job/17998742760?pr=26) by DocFx. 

I'm just curious if it will be supported by DocFx any time soon.

**Comment by KalleOlaviNiemitalo** [comment]
<https://github.com/dotnet/docfx/pull/9122> upgraded the Microsoft.CodeAnalysis.\* packages to 4.7.0, which is currently the highest stable version.  There are prereleases 4.8.0-1.final and 4.8.0-2.final.  I assume that

- the 4.8.0-* prereleases would support C# 12,
- stable Roslyn 4.8.0 packages will be published as part of the .NET 8 release, and
- the package references in DocFX will be upgraded after stable versions are available.

**Reply by KalleOlaviNiemitalo** [reply]
<https://github.com/dotnet/roslyn/blob/main/docs/wiki/NuGet-packages.md> doesn't mention version 4.8 or C# 12 yet.

**Reply by KalleOlaviNiemitalo** [reply]
.NET 8 was released but there aren't stable Roslyn packages for that yet; the latest [Microsoft.CodeAnalysis.CSharp 4.8.0-3.final](https://www.nuget.org/packages/Microsoft.CodeAnalysis.CSharp/4.8.0-3.final) is still a prerelease.

**Comment by aalmada** [comment]
Thank you!
